### ISSUE TITLE
Node compatibility issue: @azure/storage-blob

### ISSUE THREAD

**Issue by onionhammer** [oven-sh/bun#4067]
### What version of Bun is running?

0.7.3

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64 (WSL)

### What steps can reproduce the bug?

1. Run azurite emulator
  ```
  docker run --rm -d -p 10000:10000 -p 10001:10001 -p 10002:10002 mcr.microsoft.com/azure-storage/azurite
  ```
2. Install @azure/storage-blob
```
bun add @azure/storage-blob
```
3. Run the following code in bun

```ts
import { ContainerClient } from "@azure/storage-blob";

const connString = "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;"

const client = new ContainerClient(connString, "faq");
await client.createIfNotExists();

const test = client.getBlockBlobClient("reviews.md");
const helloBlob = new TextEncoder().encode("hello world");

await test.uploadData(helloBlob);
await test.download(); // <-- fails here

process.exit(0);
```

### What is the expected behavior?

Should output the first container found, then output the name of the first blob found with its content

### What do you see instead?

16824 |     }
16825 |     _read() {
16826 |         this.source.resume();
16827 |     }
16828 |     setSourceEventHandlers() {
16829 |         this.source.on("data", this.sourceDataHandler);
            ^
TypeError: this.source.on is not a function. (In 'this.source.on("data", this.sourceDataHandler)', 'this.source.on' is undefined)
      at setSourceEventHandlers (/home/user/projects/ai-inbox-helper/node_modules/@azure/storage-blob/dist/index.js:16829:8)
      at new RetriableReadableStream (/home/user/projects/ai-inbox-helper/node_modules/@azure/storage-blob/dist/index.js:16823:8)
      at new BlobDownloadResponse (/home/user/projects/ai-inbox-helper/node_modules/@azure/storage-bnlob/dist/index.js:16869:34)
      at /home/user/projects/ai-inbox-helper/node_modules/@azure/storage-blob/dist/index.js:19439:19

### Additional information

The code partially works, it's not until `blobClient.download()` that the code fails.

Update: Simplified repro so no azure sub needed

**Comment by onionhammer** [comment]
Verified issue is still occurring in 0.8.1

> 16824 |     }
16825 |     _read() {
16826 |         this.source.resume();
16827 |     }
16828 |     setSourceEventHandlers() {
16829 |         this.source.on("data", this.sourceDataHandler);
            ^
TypeError: this.source.on is not a function. (In 'this.source.on("data", this.sourceDataHandler)', 'this.source.on' is undefined)
      at setSourceEventHandlers (/home/user/projects/repro-bun-issue/node_modules/@azure/storage-blob/dist/index.js:16829:8)
      at new RetriableReadableStream (/home/user/projects/repro-bun-issue/node_modules/@azure/storage-blob/dist/index.js:16823:8)
      at new BlobDownloadResponse (/home/user/projects/repro-bun-issue/node_modules/@azure/storage-blob/dist/index.js:16869:34)
      at /home/user/projects/repro-bun-issue/node_modules/@azure/storage-blob/dist/index.js:19439:19

**Comment by onionhammer** [comment]
@Jarred-Sumner this appears to be fixed in 1.0, not sure what to attribute it to

**Comment by Chelo19** [comment]
I solved it just by upgrading @azure/storage-blob to last version (current version for me is 12.26.0) with: 

`npm i @azure/storage-blob@12.26.0`

From [npmjs.com](https://www.npmjs.com/package/@azure/storage-blob/v/12.26.0)
